src/xtopcom/xpreprocess/xpreprocess.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023-present Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
+#include "xbase/xns_macro.h"
+#include "xdata/xpreprocess_data.h"
+
+#include <memory>
+#include <string>
+
+NS_BEG2(top, data)
+class xpreprocess {
+public:
+    xpreprocess();
+    ~xpreprocess() = default;
+    static xpreprocess & instance();
+    bool send(xmessage_t & msg);
+    // if the value is false,  the following logic can continue.
+    bool async_send(xmessage_t & msg);
+private:
+    std::shared_ptr<xhandler_face_t> m_handler;
```

Comment:
why `std::shared_ptr`? `does std::unique_ptr` work or not? any special reason for shared_ptr?